src/main/java/org/testng/TestNG.java

```diff
@@ -947,9 +947,11 @@ private void addReporter(Class<? extends IReporter> r) {
   }
 
   private void initializeDefaultListeners() {
-    m_testListeners.put(ExitCodeListener.class, new ExitCodeListener(this));
+    ExitCodeListener exitCodeListener = new ExitCodeListener(this);
+    m_testListeners.put(ExitCodeListener.class, exitCodeListener);
```

Comment:
Why do you put it directly in the map instead of `addListener`?

Reply:
I think that's existing code. Not part of my changes.